bp/_io.pyx

```diff
@@ -36,6 +36,52 @@ cdef void _set_node_metadata(np.uint32_t ptr, unicode token,
     lengths[ptr] = length
 
 
+def write_newick(BP tree, object output):
+    cdef:
+        list name_stack
+        list length_stack
+        list open_paren_stack
+        object name
+        np.npy_float64 length
+        Py_ssize_t idx
+        np.npy_uint8 v
+        Py_ssize_t root_close
+
+    length_stack = []
+    name_stack = []
+    open_paren_stack = []
+    root_close = tree.close(0)
+
+    for idx, v in enumerate(tree.B):
+        if v:
+            if not tree.isleaf(idx):
+                output.write('(')
+            name_stack.append(tree.name(idx))
+            length_stack.append(tree.length(idx))
+            open_paren_stack.append(idx)
+        else:
+            name = name_stack.pop()
+            length = length_stack.pop()
+
+            if name is not None:
+                # if we have magical characters, make sure we quote
+                if set(name) & {';', ',', '(', ')'}:
```

Comment:
Can ':' be a magical character?

Reply:
oh snarky, ya good point

Reply:
Looks good. There might be other characters that we could check but I think those are the ones with special meaning behind them. If this becomes an issue we can probably do something similar along the lines of .isalnum()

Reply:
+1